### DISCUSSION TITLE
Why does firejail fail to execute if ~/.asoundrc is a symlink?

### DISCUSSION THREAD

**Discussion by amano-kenji** [netblue30/firejail#5709] [Q&A] [answered]
If ~/.asoundrc is a symlink, I see this error message.
```
Error: Cannot access /home/user/.asoundrc
Error: proc 29441 cannot sync with peer: unexpected EOF
```
Why?

**Comment by rusty-snake** [comment]
Where does it point to?
Which commands do you use?

**Reply by amano-kenji** [reply]
I found another error.
```
Error: .asoundrc is a symbolic link pointing to a file outside home directory
```

**Reply by amano-kenji** [reply]
I hate this restriction.

**Reply by amano-kenji** [reply]
Is there any elegant configuration management system?

**Reply by rusty-snake** [reply]
What do you mean with "configuration management system"?

**Reply by amano-kenji** [reply]
I chose ansible as my simple configuration management system for synchronizing dotfiles between my home directory and a git repository.

~/.asoundrc can go out of sync between directories without configuration management system like ansible.

**Reply by amano-kenji** [reply]
Do we need to restrict a symlink in HOME from pointing to a file outside home directory?

**Comment by rusty-snake** [comment]
Did you search for it? 

Is it the same as #4529 (#4527)?

**Comment by smitsohu** [comment]
It looks like Firejail only answers half of the question at the moment.

It is probably not wrong to reject copying a file from outside $HOME to a tmpfs mounted on $HOME, but if Firejail does not copy the file it could perhaps just recreate the symbolic link? If .asoundrc is under control of the user already,  it doesn't make a difference which symlink is created by Firejail. If .asoundrc is not under control of the user, Firejail could just 1:1 reproduce the symbolic link.

> I hate this restriction.

Understood.

**Comment by netblue30** [comment]
I'll remove the restriction, so you can link wherever, as long as the file pointed by the symlink is owned by the user running the sandbox.

Any other similar files/problems, just let us know.

**Reply by amano-kenji** [reply]
Thank you.

**Reply by netblue30** [reply]
Should be fixed now on mainline, you can try it out.

https://github.com/netblue30/firejail/commit/66ad14e7868d524d8b7d659e29d00cb542b6fe19
https://github.com/netblue30/firejail/commit/66f8107330d3e1ebea6566f4288b8f41d8f36621

**Reply by amano-kenji** [reply]
Why do you have code specifically for `~/.asoundrc`?

**Reply by amano-kenji** [reply]
Anyway, I tried it. It works.

**Reply by netblue30** [reply]
There are actually 3 of them: .asoundrc, .Xauthority, and .config/pulse. We also bring in from /etc/skel a .bashrc, or a .zshrc, or a .cshrc. And after all this is done we generate a new .inputrc. It looks like this:
`````
$ firejail --private --quiet
$ ls -al
total 12
drwx------ 3 netblue netblue  140 Mar  9 22:59 .
drwxr-xr-x 3 nobody  nogroup   60 Mar  9 22:59 ..
-rw------- 1 netblue netblue    0 Mar  9 22:59 .asoundrc
-rw-r--r-- 1 netblue netblue 3526 Mar  9 22:59 .bashrc
drwx------ 3 netblue netblue   60 Mar  9 22:59 .config
-rw-r--r-- 1 netblue netblue   26 Mar  9 22:59 .inputrc
-rw------- 1 netblue netblue  155 Mar  9 22:59 .Xauthority
`````

**Comment by netblue30** [comment, marked as answer]
> Anyway, I tried it. It works.

OK, thanks. I'll mark it as resolved on mainline.
